### ISSUE TITLE
Only allow GCI students to claim issues with "initiative/gci" label

### ISSUE THREAD

**Issue by co-robo** [coala/corobo#306]
How to determine who are GCI students TBD.
Opened by @jayvdb at [gitter/coala/gci](https://gitter.im/coala/gci?at=59e74be881cd9a5d7e9f6a56)

**Comment by jayvdb** [comment]
Three options.

1. Create a GCI team and populate it manually.
2. link GitHub identity with GCI system, which is not easy as it dont provide any serious identity for students in their API , only 'display name'
3. Allow assignment of GCI issues that have been claimed in GCI - that is pretty easy to do with their API.

**Comment by jayvdb** [comment]
@andrewda helped us find another related problem.
The GCI tasks include "medium" difficulty tasks, and the bot also tells them no, because they havent finished the https://coala.io/newcomer process (yet).

**Comment by meetmangukiya** [comment]
Actually medium level is developers only

**Comment by jayvdb** [comment]
Right, but that doesnt work when we have medium and hard level tasks in GCI, and we dont have 'do code review' tasks for GCI participants.

**Comment by prakashdanish** [comment]
Does having a team solve this issue? Can we have different rules for corobo for different teams?

**Comment by jayvdb** [comment]
Yea, that was option 1 above -- I didnt explain it very well ;-)

**Comment by LifetimeAlone** [comment]
Maybe it's possible to do something using GCI API?
Participants may post their GitHub profile link in task interface comments, and corobo may automatically assign issues to them.

**Comment by margobra8** [comment]
@LifetimeAlone I think that associating each student that claims a task published in GCI to the corresponding issue is very time consuming and difficult because it could run into some problems:

- The issue might be closed or fixed by another contributor (as they are opened to public contributions as well as GCI), rendering the GCI-issue connection unusable.
- Abandoning a task and then another student claiming it could run into problems with the association of the assignments. This could be circumvented by checking every time the association of the student and the issue but this would be very hard to implement.
- GCI API _still_ doesn’t have a robust way of checking a student’s identity. i.e. if I change my name to someone else’s I could be assigned the issue.

**Comment by Makman2** [comment]
> as they are opened to public contributions as well as GCI

No only to GCI students. Issues are tagged with `initiatives/gci` and public contributors can't get assigned to them.

> Abandoning a task and then another student claiming it could cause problems to the association of the assignments. This could be circumvented by checking repeatedly the association of the student and the issue but this would be very hard to implement and very inefficient.

If the google API supports sending events, then we don't have this problem. And also checking assignments every 1/5 minutes is imo not really inefficient ;) A performance bottleneck is something different ;)

**Comment by margobra8** [comment]
@Makman2 Thanks for pointing out. I didn’t know that issues tagged with `initiatives/gci` were restricted to GCI participants.
And also the performance bottleneck was the word I was looking for instead of “inefficient” thanks for mentioning it (English is not my mother tongue).

**Comment by Makman2** [comment]
> English is not my mother tongue

No worries ;)

However making a regular timely-based check is imo no problem from performance-side.

**Comment by EverWinter23** [comment]
I would like to work on this issue.

**Comment by EverWinter23** [comment]
@jayvdb @Makman2 From what I've read so far, I'll be tackling the issue the following way:
1. A separate team of GCI members.
2. If member belongs to this team and assign the issue. (keeping checks in mind)

**Comment by Makman2** [comment]
Sounds okay, not sure introducing a GCI team on GitHub would impose a problem, but I guess it doesn't. In the end we can still change how we gather together all GCI participants :)

**Comment by EverWinter23** [comment]
Thanks. Will start working on it right away.

**Comment by RaiVaibhav** [comment]
> In the end we can still change how we gather together all GCI participants :)

A `yml` file or some data file where each GCI participant have to first add their username then they can start there work?

**Comment by RaiVaibhav** [comment]
Creating a team should not have to an option here IMO
